.github/workflows/main.yml

```diff
@@ -59,39 +51,23 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
 
       - if: steps.step_conditionals_handler.outputs.is_push_to_master == 'true'
-        name: Get assembly version from appxmanifest
-        id: get_assembly_version
-        shell: pwsh
+        name: Bump GitHub tag
+        id: tag_generator
+        uses: mathieudutour/github-tag-action@v5
+        with: 
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          default_bump: false
+
+      - if: steps.step_conditionals_handler.outputs.is_push_to_master == 'true' && steps.tag_generator.outputs.new_version != ''
+        name: Update Package.appxmanifest version
+        id: update_appxmanifest
         run: |
-          cd src/Notepads/
-          $xml = [xml](Get-Content Package.appxmanifest)
-          $ASSEMBLY_VERSION_NUMBER = $xml.Package.Identity | Select -ExpandProperty Version
-          echo "::set-output name=version_num::$(echo $ASSEMBLY_VERSION_NUMBER)"
-          echo "::set-output name=version_tag::$(echo v"$ASSEMBLY_VERSION_NUMBER")"
-
-      - if: steps.step_conditionals_handler.outputs.is_push_to_master == 'true'
-        name: Get latest tag
-        id: get_latest_tag
-        shell: pwsh
-        run: |
-          $LATEST_TAG = git -c 'versionsort.suffix=-' ls-remote --exit-code --refs --sort='version:refname' --tags "https://github.com/$env:GIT_URL.git" '*.*.*' | tail --lines=1 | cut --delimiter='/' --fields=3
-          echo "::set-output name=tag::$(echo $LATEST_TAG)"
+          $APPXMANIFEST_PATH = 'src/Notepads/Package.appxmanifest'
+          $xml = [xml](Get-Content $APPXMANIFEST_PATH)
+          $xml.Package.Identity.SetAttribute('Version', "$env:NEW_VERSION.0")
+          $xml.save($APPXMANIFEST_PATH)
```

Comment:
@aleks-ivanov This step will not change the package version in the manifest file in repo, right?? We have to manually change the original manifest to keep things in sync??

Reply:
@soumyamahunt it **does** change it, because the new version as I understand is needed for the packaging process.

Reply:
@soumyamahunt would like the `Package.appxmanifest` file to be committed to the repo with the updated version ?

Reply:
> @soumyamahunt would you like the `Package.appxmanifest` file to be committed to the repo with the updated version ?

It's not necessary since we are using previous tag to genarate new version regardless of what manifest file version number. Ignore my previous comment.